### PR TITLE
Add emotion detection

### PR DESCRIPTION
**Pull request by Steffy-zxf**
Add emotion detection

### CODE REVIEW

File: modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# -*- coding:utf-8 -*-
+import os
+import io
+import platform
+
+import six
+import paddle
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py]: 冗余import

File: modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# -*- coding:utf-8 -*-
+import os
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py]: 冗余import

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
+# -*- coding:utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import ast
+import json
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 冗余import

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+    def __init__(self, *args):
+        self.args = args
+
+
+@moduleinfo(
+    name="emotion_detection_textcnn",
+    version="1.1.0",
+    summary="Baidu's open-source Emotion Detection Model(TextCNN).",
+    author="baidu-nlp",
+    author_email="paddle-dev@baidu.com",
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 这里建议置空

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+    type="nlp/sentiment_analysis")
+class EmotionDetectionTextCNN(hub.Module):
+    def _initialize(self, ):
+        """
+        initialize with the necessary elements
+        """
+        self.pretrained_model_path = os.path.join(self.directory, "infer_model")
+        self.vocab_path = os.path.join(self.directory, "assets/vocab.txt")
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 正确的写法应该为
```python
os.path.join(self.directory, "assets", "vocab.txt")
```

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+
+@moduleinfo(
+    name="emotion_detection_textcnn",
+    version="1.1.0",
+    summary="Baidu's open-source Emotion Detection Model(TextCNN).",
+    author="baidu-nlp",
+    author_email="paddle-dev@baidu.com",
+    type="nlp/sentiment_analysis")
+class EmotionDetectionTextCNN(hub.Module):
+    def _initialize(self, ):
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 多余的逗号去除

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+        elif texts == [] and isinstance(data, dict) and isinstance(
+                data.get('text', None), list) and data['text']:
+            predicted_data = data["text"]
+        else:
+            raise ValueError(
+                "The input data is inconsistent with expectations.")
+
+        predicted_data = self.to_unicode(predicted_data)
+        if not self.lac:
+            self.lac = hub.Module(name='lac')
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 建议使用这样的代码实现
```python
def _initialize(self):
    self._word_seg_module = None

@property
def word_seg_module(self):
    if not self._word_seg_module:
        self._word_seg_module = hub.Module(name="lac")
    return self._word_seg_module

File: modules/text/sentiment_analysis/emotion_detection_textcnn/net.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        act="softmax",
+        param_attr=fluid.ParamAttr(
+            name="@HUB_emotion_detection_textcnn@fc_1.w_0"),
+        bias_attr=fluid.ParamAttr(
+            name="@HUB_emotion_detection_textcnn@fc_1.b_0"))
+    cost = fluid.layers.cross_entropy(input=prediction, label=label)
+    avg_cost = fluid.layers.mean(x=cost)
+    acc = fluid.layers.accuracy(input=prediction, label=label)
+
+    return avg_cost, acc, prediction, fc_1
```
[the remark this answers is not in the document]
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/net.py]: 已去掉

File: modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+import io
+import platform
```
[the remark this answers is not in the document]
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py]: 已去掉

File: modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+import six
```
[the remark this answers is not in the document]
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py]: 已去掉

File: modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+import paddle
+import paddle.fluid as fluid
+import numpy as np
+
+import paddlehub as hub
```
[the remark this answers is not in the document]
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py]: 已去掉

File: modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py [hunk 12]
```diff
@@ -0,0 +1,76 @@
+# -*- coding:utf-8 -*-
+import os
+import io
+import platform
+
+import six
+import paddle
+import paddle.fluid as fluid
```
[the remark this answers is not in the document]
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py]: 已去掉

File: modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py (hunk 1 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py]: 已去掉

File: modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py (hunk 2 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/processor.py]: 已去掉

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py (hunk 3 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 已去掉

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py (hunk 4 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 已修改

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py (hunk 5 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 已修改

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py (hunk 6 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 已去除

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py (hunk 7 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 已修改

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py [hunk 13]
```diff
@@ -0,0 +1,308 @@
...
+            predicted_data = texts
+        elif texts == [] and isinstance(data, dict) and isinstance(
+                data.get('text', None), list) and data['text']:
+            predicted_data = data["text"]
+        else:
+            raise ValueError(
+                "The input data is inconsistent with expectations.")
+
+        predicted_data = self.to_unicode(predicted_data)
+        processed_results = preprocess(self.word_seg_module, predicted_data,
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 应该同时支持batch_size的输入

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py [hunk 14]
```diff
@@ -0,0 +1,308 @@
...
+
+        args = self.parser.parse_args(argvs)
+
+        try:
+            input_data = self.check_input_data(args)
+        except DataFormatError and RuntimeError:
+            self.parser.print_help()
+            return None
+
+        results = self.classify(
```
Comment (nepeplwu) [inline modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 不存在这个接口

Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 已修改

File: modules/text/sentiment_analysis/emotion_detection_textcnn/module.py (hunk 13 above)
Comment (Steffy-zxf) [reply modules/text/sentiment_analysis/emotion_detection_textcnn/module.py]: 已修改
